components/scream/src/physics/p3/p3_functions_ice_deposition_sublimation_impl.hpp

```diff
@@ -12,13 +12,14 @@ KOKKOS_FUNCTION
 void Functions<S,D>
 ::ice_deposition_sublimation(const Spack& qitot_incld, const Spack& nitot_incld, const Spack& t,   const Spack& qvs,
 			     const Spack& qvi,         const Spack& epsi,        const Spack& abi, const Spack& qv,
-			     Spack& qidep, Spack& qisub, Spack& nisub, Spack& qiberg)
+			     Spack& qidep, Spack& qisub, Spack& nisub, Spack& qiberg,
+                             const Smask& context)
 {
   constexpr Scalar QSMALL   = C::QSMALL;
   constexpr Scalar ZERODEGC = C::ZeroDegC;
   const auto oabi           = sp(1.0)/abi;
 
-  const auto qitot_incld_not_small = qitot_incld >= QSMALL;
+  const auto qitot_incld_not_small = qitot_incld >= QSMALL && context;
```

Comment:
Given that you're negating this and &&ing it with `context` below, it may be easier to analyze if you just use `qitot_incld && context` in line 25.

Reply:
I would have to to add `&& context` to another 10 lines or so if I did that. The approach I took with these context masks is to `&&` them into masks already being used if possible to minimize the ugliness. You have to be careful when doing this if the code contains notted (`!`) masks, since this would potentially turn on pack indices that should be inactive.